### PR TITLE
feature(LifetimeBogo): add new lifetime buy one get one product to intersection

### PR DESCRIPTION
**Pull request by balintd**
Closes https://languagedrops.atlassian.net/browse/DR-2230

### CODE REVIEW

File: dist/version-4/purchaseUtils.js [hunk 1]
```diff
@@ -286,6 +289,8 @@ const getProductIdMobile = (identifier, platform, appType) => {
             }
         case PayloadIAPIdentifierV4.iapPremiumLifetime180:
             return `premium_lifetime_180_${iapSuffix.replace('lang_', '')}_v2`;
+        case PayloadIAPIdentifierV4.iapPremiumLifetimeBogo180:
+            return `premium_lifetime_bogo_180_${iapSuffix.replace('lang_', '')}_v2`;
```
Comment (golendukhin) [inline dist/version-4/purchaseUtils.js:293]: @balintd do you think we need string mutations here or just using something like `premium_lifetime_bogo_180_${iapSuffix}` would be enough?

File: src/version-4/purchaseUtils.ts [hunk 2]
```diff
@@ -297,6 +301,8 @@ const getProductIdMobile = (identifier: PayloadIAPIdentifierV4, platform: AppPla
       }
     case PayloadIAPIdentifierV4.iapPremiumLifetime180:
       return `premium_lifetime_180_${iapSuffix.replace('lang_', '')}_v2`
+    case PayloadIAPIdentifierV4.iapPremiumLifetimeBogo180:
+      return `premium_lifetime_bogo_180_${iapSuffix.replace('lang_', '')}_v2`
```
Comment (golendukhin) [inline src/version-4/purchaseUtils.ts:305]: same question here

File: dist/version-4/purchaseUtils.js (hunk 1 above)
Comment (balintd) [reply dist/version-4/purchaseUtils.js:293]: @golendukhin Yes, I believe we need this for generating an IAP for every niche app on Android, not just for int.

Comment (golendukhin) [reply dist/version-4/purchaseUtils.js:293]: I mean why do we need to remove 'lang_' here and also 'v2'?

**Pull request closed without merging**